packages/plugin-e2e/src/e2e-selectors/type-testing/guards.ts

```diff
@@ -0,0 +1,14 @@
+import { VersionedSelector, VersionedSelectorGroup } from './types';
+import { keysOf } from './utils';
+
+export function isVersionedSelectors(value: unknown): value is VersionedSelector {
+  const notSelectorFunction = keysOf(value).find((key) => typeof value[key] !== 'function');
+  return !Boolean(notSelectorFunction);
+}
```

Comment:
maybe something along those lines?
```typescript
return keysOf(value).every((key) => typeof value[key] === 'function')
```

Reply:
but hmm..that doesn't take strings into account. 

return keysOf(value).every((key) => typeof value[key] === 'function') || keysOf(value).every((key) => typeof value[key] === 'string')